docs/interactions/Receiving_and_Responding.md

```diff
@@ -0,0 +1,314 @@
+| PING                | 1     |
+| APPLICATION_COMMAND | 2     |
+| MESSAGE_COMPONENT   | 3     |
+
+###### Interaction Data Structure
+
+| Field          | Type                                                                                                                                                                                   | Description                                                                                                                                         |
+| -------------- | -------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------- |
+| id             | snowflake                                                                                                                                                                              | for slash commands, the [`ID`](#DOCS_INTERACTIONS_SLASH_COMMANDS/application-command-object-application-command-structure) of the invoked command   |
+| name           | string                                                                                                                                                                                 | for slash commands, the [`name`](#DOCS_INTERACTIONS_SLASH_COMMANDS/application-command-object-application-command-structure) of the invoked command |
+| resolved?      | [application command interaction data resolved](#DOCS_INTERACTIONS_SLASH_COMMANDS/application-command-interaction-object-application-command-interaction-data-resolved-structure)      | for slash commands, converted users + roles + channels                                                                                              |
+| options?       | array of [application command interaction data option](#DOCS_INTERACTIONS_SLASH_COMMANDS/application-command-interaction-object-application-command-interaction-data-option-structure) | for slash commands, the params + values from the user                                                                                               |
+| custom_id      | string                                                                                                                                                                                 | for components, the [`custom_id`](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/custom-id) of the component                                                 |
+| component_type | integer                                                                                                                                                                                | for components, the [type](#DOCS_INTERACTIONS_MESSAGE_COMPONENTS/component-types) of the component                                                  |
```

Comment:
Awesome. I got a little bit of feedback on this section, splitting it up per interaction type (eliminating that for slash commands, for components, for some future interaction stuff). Any thoughts on this?

Reply:
Commented below on this